runtime/src/main/java/com/tietoevry/quarkus/resteasy/problem/javax/Violation.java

```diff
@@ -1,29 +1,32 @@
 package com.tietoevry.quarkus.resteasy.problem.javax;
 
-import java.util.Locale;
-
 public final class Violation {
 
     public enum In {
-        QUERY,
-        PATH,
-        HEADER,
-        FORM,
-        BODY;
+        query,
+        path,
+        header,
+        form,
+        body,
+        unknown;
 
-        public Violation violation(String message, String field) {
-            return new Violation(message, field, this);
+        public MessageSupplier violation(String field) {
```

Comment:
Maybe I miss something but this is a bit over-engineered for me.

Reply:
Taking into consideration fact that this constructor takes arguments `String`, `String` it is very easy to provide wrong arguments in the wrong place which could lead to incorrect Violation object.
It's another piece of complexity to achieve simplicity :)

Reply:
Builder?

Reply:
Yeap, it's an enhanced builder, which does not allow you to build an object that is broken/not finished. It will give you constructed object only if you provide all mandatory fields, which I found a great value for the price of more complicated builder implementation :)

I improved method names a bit (violation -> field, message -> withMessage()